cmd/extract/main.go

```diff
@@ -28,27 +28,34 @@ const (
 	typeFullBackup = 2
 )
 
+var debug = os.Getenv("DEBUG") == "1"
+
 func main() {
 	if len(os.Args) != 3 {
 		_, _ = fmt.Fprintf(os.Stderr, "usage: %s path-to-backup mnemonic\n", filepath.Base(os.Args[0]))
 		os.Exit(1)
 	}
 
 	backupPath := os.Args[1]
+	err := extractAppBackup(backupPath)
+	if err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "error: %s\n", err)
+		os.Exit(1)
+	}
+}
+
+func extractAppBackup(backupPath string) error {
 	metadataPath := filepath.Join(backupPath, ".backup.metadata")
 	_, err := os.Stat(metadataPath)
 	if errors.Is(err, os.ErrNotExist) {
-		_, _ = fmt.Fprintln(os.Stderr, "error: not a backup (missing .backup.metadata)")
-		os.Exit(1)
+		return fmt.Errorf("not a backup (missing .backup.metadata)")
 	} else if err != nil {
-		_, _ = fmt.Fprintf(os.Stderr, "error: failed to stat %q: %s\n", metadataPath, err)
-		os.Exit(1)
+		return fmt.Errorf("failed to stat %q: %s\n", metadataPath, err)
```

Comment:
Can we remove the newlines from any returned errors? We're ending up with an extra one for some errors now as we're also adding a newline when printing out any extraction errors.